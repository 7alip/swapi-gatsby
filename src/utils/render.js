import React from "react"
import { render as rtlRender } from "@testing-library/react"
import { ThemeProvider } from "@chakra-ui/core"

export function render(container) {
  return rtlRender(<ThemeProvider>{container}</ThemeProvider>)
}
