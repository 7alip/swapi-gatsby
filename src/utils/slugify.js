const slugify = require("slugify")

exports.slug = text =>
  slugify(text, {
    replacement: "-",
    strict: true,
    lower: true,
  })
