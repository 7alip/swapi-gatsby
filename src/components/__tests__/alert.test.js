import React from "react"
import renderer from "react-test-renderer"
import { cleanup } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import { render } from "../../utils/render"
import Alert from "../alert"
import { ThemeProvider } from "@chakra-ui/core"

afterEach(cleanup)

const renderComponent = () => {
  return render(<Alert status="error" title="Test Title" text="Test text" />)
}

describe("Alert", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <ThemeProvider>
        <Alert status="error" title="Test Title" text="Test text" />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render", () => {
    renderComponent()
  })

  it("should display title", () => {
    const { getByText } = renderComponent()

    expect(getByText(/test title/i)).toBeInTheDocument()
  })

  it("should display text", () => {
    const { getByText } = renderComponent()

    expect(getByText(/test text/i)).toBeInTheDocument()
  })
})
