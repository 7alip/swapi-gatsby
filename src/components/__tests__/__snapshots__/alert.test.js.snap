// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot renders 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  overflow: hidden;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fed7d7;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  color: #e53e3e;
  display: inline-block;
  vertical-align: middle;
  margin-right: 0.75rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-0:not(:root) {
  overflow: hidden;
}

.emotion-1 {
  font-weight: 700;
  line-height: normal;
  margin-right: 0.5rem;
}

.emotion-4 {
  outline: none;
  width: 32px;
  height: 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: absolute;
  right: 8px;
  top: 8px;
}

.emotion-4:hover {
  background-color: rgba(0,0,0,0.06);
}

.emotion-4:focus {
  box-shadow: 0 0 0 3px rgba(66,153,225,0.6);
}

.emotion-4:active,
.emotion-4[data-active=true] {
  background-color: rgba(0,0,0,0.08);
}

.emotion-4:disabled,
.emotion-4:disabled:focus,
.emotion-4:disabled:hover,
.emotion-4[aria-disabled=true],
.emotion-4[aria-disabled=true]:focus,
.emotion-4[aria-disabled=true]:hover {
  cursor: not-allowed;
}

.emotion-3 {
  width: 12px;
  height: 12px;
  color: currentColor;
  display: inline-block;
  vertical-align: middle;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.emotion-3:not(:root) {
  overflow: hidden;
}

<div
  className="emotion-5"
  role="alert"
>
  <svg
    className="emotion-0"
    focusable={false}
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
      fill="currentColor"
    />
  </svg>
  <div
    className="emotion-1"
  >
    Test Title
  </div>
  <div
    className="emotion-2"
  >
    Test text
  </div>
  <button
    aria-label="Close"
    className="emotion-4"
    type="button"
  >
    <svg
      aria-hidden={true}
      className="emotion-3"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
